pkg/datapath/iptables/iptables.go

```diff
@@ -327,6 +336,26 @@ func (m *IptablesManager) InstallRules(ifName string) error {
 			return err
 		}
 
+		// For comminucation from host to local services via k8s cluster IP
+		// we need to fix up wrong source address selection. Linux routing
+		// will initially select a source address based on the service IP and
+		// after Kubernetes iptables rules selected a local backend, we still
+		// retain the original source address (and not the one on cilium_host).
+		// As a result, ipcache will assign a WORLD identity as opposed to a
+		// HOST identity and therefore policy is dropping the skb. As there
+		// is no fixup by iptables, we need to SNAT for these cases. This rule
+		// here must come after all Kubernetes post-routing chains, so we can
+		// match on our endpoint allocation range.
+		if err := runProg("iptables", []string{
+			"-t", "nat",
+			"-A", ciliumPostNatKubeChain,
```

Comment:
Can we add `"!", "-s", node.GetInternalIPv4().String()` to prevent from unnecessary SNAT'ing in the case someone has installed a route (`cilium_host` -> k8s service subnet) manually?

Reply:
Hm, but wouldn't this potentially collide since it's the same address we're SNATing to?

Reply:
Not sure about the collision. I was thinking about a case when a user installs the route manually to work around the problem. Anyway, the proposed `! -s <...>` is probably an unnecessary complication.